hil-test/tests/embassy_interrupt_spi_dma.rs

```diff
@@ -128,7 +129,7 @@ mod test {
                 ..Config::default()
             },
         )
-        .with_miso(miso)
+        .with_miso(unsafe { mosi.clone_unchecked() })
```

Comment:
you can `let (miso, mosi) = mosi.split()` instead safely.

Reply:
I wanted to do that / tried that but
```
error[E0277]: the trait bound `esp_hal::gpio::interconnect::InputSignal: PeripheralOutput` is not satisfied
   --> tests\embassy_interrupt_spi_dma.rs:133:10
    |
133 |         .with_miso(miso)
    |          ^^^^^^^^^ the trait `OutputPin` is not implemented for `esp_hal::gpio::interconnect::InputSignal`, which is required by `esp_hal::gpio::interconnect::InputSignal: PeripheralOutput`
    |
    = help: the following other types implement trait `OutputPin`:
              AnyPin
              GpioPin<0>
              GpioPin<10>
              GpioPin<11>
              GpioPin<12>
              GpioPin<13>
              GpioPin<14>
              GpioPin<15>
            and 38 others
    = note: required for `esp_hal::gpio::interconnect::InputSignal` to implement `PeripheralOutput`
note: required by a bound in `esp_hal::spi::master::Spi::<'d, M, T>::with_miso`
   --> D:\projects\esp\esp-hal\esp-hal\src\spi\master.rs:625:28
    |
625 |     pub fn with_miso<MISO: PeripheralOutput>(self, miso: impl Peripheral<P = MISO> + 'd) -> Self {
    |                            ^^^^^^^^^^^^^^^^ required by this bound in `Spi::<'d, M, T>::with_miso`

```

Why is master-in `Output`?

Reply:
Whoops, right. It's Output because the driver also supports half-duplex.